### PR TITLE
IGNITE-20972 Move handlers of zone-related commands from CatalogManager

### PR DESCRIPTION
**Pull request by zstan**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Review by ygerzhedovich — APPROVED** [review]

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogManagerImpl.java [hunk 1]
```diff
@@ -337,97 +322,6 @@ public CompletableFuture<Void> execute(List<CatalogCommand> commands) {
...
-            validateAlterZoneParams(params);
-
-            CatalogZoneDescriptor zone = getZone(catalog, params.zoneName());
-
-            CatalogZoneDescriptor descriptor = fromParamsAndPreviousValue(params, zone);
-
-            return List.of(new AlterZoneEntry(descriptor));
-        });
-    }
-
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogManagerImpl.java:430]: `org.apache.ignite.internal.catalog.CatalogManagerImpl#getZone` can be removed

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogParamsValidationUtils.java [hunk 2]
```diff
@@ -49,95 +44,27 @@ public static void validateIdentifier(@Nullable String identifier, String contex
...
+    public static void validateField(@Nullable Integer value, int min, @Nullable Integer max, String errorPrefix) {
+        if (value == null) {
+            return;
+        }
 
-    private static void validateZoneDataNodesAutoAdjustScaleDown(@Nullable Integer dataNodesAutoAdjustScaleDown) {
-        validateZoneField(dataNodesAutoAdjustScaleDown, 0, null, "Invalid data nodes auto adjust scale down");
+        if (value < min || (max != null && value > max)) {
+            throw new CatalogValidationException(
+                    DistributionZones.ZONE_DEFINITION_ERR,
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogParamsValidationUtils.java:57]: let's throw code from catalog's error group ()or just Catalog.VALIDATION_ERR. The same for rest of validation exception

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractZoneCommand.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import static org.apache.ignite.internal.catalog.CatalogParamsValidationUtils.validateIdentifier;
+
+import org.apache.ignite.internal.catalog.CatalogCommand;
+import org.apache.ignite.internal.catalog.CatalogValidationException;
+
+/**
+ * Abstract zone-related command.
+ *
+ * <p>Every table-related command, disregard it going to create new table or modify existing one,
+ * should specify name of the table and namespace (schema) where to find existing/put new table.
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractZoneCommand.java:29]: I see this is copy-pasted from table's command... Let's adjust javadoc accordingly

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractZoneCommandBuilder.java [hunk 4]
```diff
@@ -17,27 +17,17 @@
...
 package org.apache.ignite.internal.catalog.commands;
 
+import org.apache.ignite.internal.catalog.CatalogCommand;
+
 /**
- * DROP ZONE statement.
+ * Abstract builder of zone-related commands.
+ *
+ * <p>Every table-related command, disregard it going to create new table or modify existing one,
+ * should specify name of the table and namespace (schema) where to find existing/put new table.
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractZoneCommandBuilder.java:26]: the same as above: javadoc should be adjusted

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RenameZoneParams.java [hunk 5]
```diff
@@ -1,68 +0,0 @@
...
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.ignite.internal.catalog.commands;
-
-/**
- * ALTER ZONE RENAME TO statement.
- */
-public class RenameZoneParams extends AbstractZoneCommandParams {
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RenameZoneParams.java:23]: why don't you remove rest of the zone-related params?

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java [hunk 6]
```diff
@@ -198,6 +199,24 @@ public static CatalogTableColumnDescriptor fromParams(ColumnParams params) {
                 precision, scale, length, defaultValue);
     }
 
+    /**
+     * Returns appropriate zone descriptor.
+     *
+     * @param catalog Catalog descriptor.
+     * @param zoneName Zone name.
+     */
+    static CatalogZoneDescriptor getZone(Catalog catalog, String zoneName) {
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:208]: there is already `zoneOrThrow` method

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AlterZoneCommandBuilder.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * Sets the number of partitions.
+     *
+     * @param partitions Number of partitions.
+     * @return This instance.
+     */
+    AlterZoneCommandBuilder partitions(@Nullable Integer partitions);
+
+    /**
+     * Sets the number of replicas.
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AlterZoneCommandBuilder.java:39]: please add restrictions to the param description to every param of Avery builder (restrictions like `Optional`, or `Should be in range 1..Int.MAX_VALUE`; in other words, anything that is being validated )

**Comment by zstan** [comment]
@korlov42 seems all fixed

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractZoneCommandBuilder.java [hunk 8]
```diff
@@ -17,27 +17,17 @@
...
-    private DropZoneParams(String zoneName) {
-        super(zoneName);
-    }
-
-    /** Creates parameters builder. */
-    public static Builder builder() {
-        return new Builder();
-    }
+public interface AbstractZoneCommandBuilder<T extends AbstractZoneCommandBuilder<T>> {
+    T zoneName(String zoneName);
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractZoneCommandBuilder.java:29]: missing javadoc

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CreateZoneCommand.java [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+        this.filter = filter;
+        this.dataStorageParams = dataStorageParams;
+
+        validate();
+    }
+
+    @Override
+    public List<UpdateEntry> get(Catalog catalog) {
+        if (catalog.zone(zoneName) != null) {
+            throw new DistributionZoneExistsValidationException("Distribution zone already exists [zoneName=" + zoneName + ']');
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CreateZoneCommand.java:107]: The patter for this type of exception is `<Object> with name '{}' already exists` (see `org.apache.ignite.internal.catalog.CatalogParamsValidationUtils#ensureNoTableIndexOrSysViewExistsWithGivenName`)

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RenameZoneCommand.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+        validate();
+    }
+
+    @Override
+    public List<UpdateEntry> get(Catalog catalog) {
+        CatalogZoneDescriptor zone = zoneOrThrow(catalog, zoneName);
+
+        if (catalog.zone(newZoneName) != null) {
+            throw new DistributionZoneExistsValidationException("Distribution zone already exists [zoneName=" + newZoneName + ']');
```
Comment (korlov42) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RenameZoneCommand.java:65]: message pattern, see comment above

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/commands/AlterZoneCommandValidationTest.java [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Tests to verify validation of {@link AlterZoneCommand}.
+ */
+@SuppressWarnings("ThrowableNotThrown")
+public class AlterZoneCommandValidationTest extends AbstractCommandValidationTest {
```
[the remark this answers is not in the document]
Comment (korlov42) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/commands/AlterZoneCommandValidationTest.java:40]: better to have one here

**Pull request merged**